File: 은지/3진법 뒤집기.py
def tri(x):
    if x < 3:
        return x
    else:
        return tri(x // 3) * 10 + x % 3

def solution(n):
    answer = 0
    # 10진법 -> 3진법
    t = str(tri(n))
    # 앞뒤 반전 -> 10진법
    # 앞에서부터 3^i씩 곱하기와 동일
    for i in range(len(t)):
        answer += int(t[i]) * 3 ** (i)
    return answer

File: 태성/3진법뒤집기.js

function solution(n) {
    var answer = 0
    var pre = []
    var n;
    while (n >= 3) {
        pre.push(n%3)
        n = Math.floor(n/3)
    }
    pre.push(n)
    console.log(pre)

    pre.reverse()
    for (let i=0; i<pre.length; i++) {
        answer += pre[i]*(Math.pow(3, i))
    }

    return answer
}

console.log(solution(45))